x-pack/plugins/canvas/server/lib/get_es_filter.ts

```diff
@@ -10,16 +10,27 @@
   filter is the abstracted canvas filter.
 */
 
-/*eslint import/namespace: ['error', { allowComputed: true }]*/
+import {
+  CanvasQueryFilter,
+  ElasticsarchTermFilter,
+  ElasticsearchLuceneQueryStringFilter,
+  ElasticsearchTimeFilter,
+} from './filters';
 import * as filters from './filters';
 
-export function getESFilter(filter) {
-  if (!filters[filter.type]) {
+interface FilterTypes {
+  [key: string]: any;
+}
+
+export function getESFilter(
+  filter: CanvasQueryFilter
+): ElasticsarchTermFilter | ElasticsearchTimeFilter | ElasticsearchLuceneQueryStringFilter {
+  if (!(filters as FilterTypes)[filter.type]) {
```

Comment:
This is really strange... this cast should not be necessary.

Reply:
Tell the linter/compiler that 😛.

Because this ends up importing everything that `filters.ts` exports using `* as`, it seemed to get confused about what that variable holds. The reality is, we don't really care here, the code is going to take a `CanvasQueryFilter` and give back an `ElasticsearchFilter` by running it through some imported function. I don't know how to be more explicit about what's happening here.